x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValuesSourceReaderOperator.java

```diff
@@ -241,6 +241,14 @@ private void loadFromSingleLeaf(Block[] blocks, int shard, int segment, BlockLoa
                 }
             }
 
+            SourceLoader sourceLoader = null;
+            if (storedFieldsSpec.requiresSource()) {
+                sourceLoader = shardContexts.get(shard).newSourceLoader.get();
+                storedFieldsSpec = storedFieldsSpec.merge(
+                    new StoredFieldsSpec(true, storedFieldsSpec.requiresMetadata(), sourceLoader.requiredStoredFields())
+                );
+            }
```

Comment:
This seems duplicated? I wonder if we can add a helper function `getSourceLoaderLeaf` to encapsulate this logic, along with getting the leaf.

Reply:
It is duplicated, but if this is turned into a helper method it isn't necessarily getting cleaner. It would need to return a source loader and stored field spec, which means a tuple or something like that needs to be used, which makes using the helper method verbose.